docs/migrating-from-0.6.md

```diff
@@ -0,0 +1,313 @@
+
+## 1. CLI Changes
+
+The command-line interface (CLI) has undergone significant updates in Genkit 0.9. The command to start Genkit has changed, and the CLI has been separated into its own standalone package, which you now need to install separately.
+
+To install the CLI:
+```
+npm i -g genkit-cli
+```
+
+**Old Command:**
+```
+genkit start
+```
+This command starts both the flow server and the Dev UI in a single command.
+
+**New Commands:**
+
+Now, you must start the dev UI and the flow servers as separate steps.
+
+To start the dev UI:
+```
+genkit ui:start
+```
+
+Once the UI is running, start the flow server:
+```
+GENKIT_ENV=dev tsx --watch path/to/index.ts
+```
+
+Starting the flow server this way makes it easier for you to attach a debugger to your code. Be sure to set `GENKIT_ENV=dev` in your debugger’s startup configuration.
+
+The Dev UI will interact directly with the flow server to figure out which flows are registered and allow you to invoke them directly with sample inputs.
+
+## 2. Simplified packages and imports
+
+Previously, the Genkit libraries were separated into several modules, which you needed to install and import individually. These modules have now been consolidated into a single import. In addition, the Zod module is now re-exported by Genkit.
+
+**Old:**
+```
+npm i @genkit-ai/core @genkit-ai/ai @genkit-ai/flow @genkit-ai/dotprompt
+```
+
+**New:**
+```
+npm i genkit
+```
+
+**Old:**
+```js
+import { … } from '@genkit-ai/ai';
+import { … } from '@genkit-ai/core';
+import { … } from '@genkit-ai/flow';
+import * as z from 'zod';
+```
+
+**New:**
+```js
+import { genkit, z } from 'genkit';
+```
+
+Genkit plugins still must be installed and imported individually.
+
+## 3. Configuring Genkit
+
+Previously, initializing Genkit was done once globally by calling the `configureGenkit` function. Genkit resources (flows, tools, prompts, etc.) would all automatically be wired with this global configuration.
+
+Genkit 0.9 introduces Genkit instances, each of which encapsulates a configuration. See the following examples:
+
+**Old:**
+
+```js
+import { configureGenkit } from '@genkit-ai/core';
+
+configureGenkit({
+  telemetry: {
+    instrumentation: ...,
+    logger: ...
+  }
+});
+```
+
+**New:**
+
+```js
+import { genkit } from 'genkit';
+import { logger } from 'genkit/logging';
+import { enableFirebaseTelemetry } from '@genkit-ai/firebase';
+
+logger.setLogLevel('debug');
+enableFirebaseTelemetry({...});
+
+const ai = genkit({ ... });
+```
+
+Let’s break it down:
+- `configureGenkit()` has been replaced with `genkit()`, and it returns a configured genkit instance rather than setting up configurations globally.
```

Comment:
```suggestion
- `configureGenkit()` has been replaced with `genkit()`, and it returns a configured `Genkit` instance rather than setting up configurations globally.
```